sherpa/ui/utils.py

```diff
@@ -13919,32 +14022,99 @@ def _multi_plot(self,
+               pars["id"].annotation == "IdType | None":
+                allow_multiple_ids = True
+            else:
+                allow_multiple_ids = False
+
             # Collect the arguments for the get_<>_plot/contour
             # call. Loop through until we hit a supported
             # plot/contour type.
             #
             getargs = []
+
+            # If the ids argument is set, add it to the call.
+            # There is no check
+            # - whether it matches the data (e.g. ids=[1,2] when
+            #   the plot type does not support multiple ids
+            # - whether an identifier is already included
+            # as these are considered user errors.
+            #
+            if ids is not None:
+                getargs.append(ids)
+
             while largs:
-                if check(largs[0]):
+                if isinstance(largs[0], str) and check(largs[0]):
                     break
 
                 getargs.append(largs.pop(0))
 
-            funcname = f"get_{plottype}_{plotmeth}"
-            getfunc = getattr(self, funcname)
+            # If the last element of getargs is a dict then remove it
+            # and add it to stores, otherwse add an empty dict. It
+            # would be nice to have some way other than the isinstance
+            # check, but it is hard, since model components (which can
+            # be included) have dict-like features.
+            #
+            if getargs and isinstance(getargs[-1], dict):
+                stores.append(getargs.pop(-1))
+            else:
+                stores.append({})
 
-            # Need to make sure we have a copy of each plot
-            # object to support plots like
+            # Need to make sure we have a copy of each plot object to
+            # support plots like
+            #
             #    plot("data", 1, "data", 2)
             #
-            plots.append(copy.deepcopy(getfunc(*getargs)))
+            # However, it is complicated when supporting multiple
+            # identifiers such as
+            #
+            #    plot("data", [1, 2], "model", [1, 2])
+            #
+            # since the "get_xxx_plot" calls do not support multiple
+            # identifiers. This logic is dependent on the code - so
+            # both how arguments are typed and what particular
+            # routines do not support multiple identifiers - so these
+            # heuristics may need updating in the future.
+            #
+            if getargs and allow_multiple_ids and \
+               is_iterable_not_str(getargs[0]):
+                def getid(id, recalc=True):
+                    return getfunc(id, *getargs[1:], recalc=True)
+
+                plotobj = self._get_plot_objects(getargs[0], getid,
+                                                 recalc=True)
+                plots.append(copy.deepcopy(plotobj))
+
+            else:
+                plots.append(copy.deepcopy(getfunc(*getargs)))
```

Comment:
I have to admit that I don't fully understand this logic, but apparently it generates plots, so it works.

Reply:
So this part is definitely annoying to support and this was the best I could come up with.